exec/java-exec/src/main/java/org/apache/drill/exec/work/WorkManager.java

```diff
@@ -86,6 +86,9 @@
   private final StatusThread statusThread;
   private final Lock isEmptyLock = new ReentrantLock();
   private final Condition isEmptyCondition = isEmptyLock.newCondition();
+  private boolean isShutdownTriggered = false;
```

Comment:
Is this boolean necessary? Can you delay getting `isEmptyCondition` till shutdown is requested and use it in place of `isShutdownTriggered`?